### PR TITLE
Fix build from changes in rapidsai/cudf#8528

### PR DESCRIPTION
**Pull request by cwharris**
rapidsai/cudf#8528 introduced some changes which broke the cuspatial build. ~specifically `cudf.DataFrame._from_data` was change to a `@classmethod`~ cudf no longer accidentally treats `_from_data` as an instance method, which is bad behavior cuspatial was depending on. To fix this, adjustments need to be made to `cuspatial.GeoDataFrame` and some associated utilities.

PR is WIP, but so far fixes all but one failing test.

### CODE REVIEW

File: python/cuspatial/cuspatial/geometry/geodataframe.py [hunk 1]
```diff
@@ -90,9 +90,30 @@ def groupby(self, *args, **kwargs):
...
+        columns = None,
+    ):
+        out = cls.__new__(cls)
+        out._data = data
+        if index is None:
+            index = cudf.Index(range(data.nrows))
+        out._index = index
+        if columns is not None:
+            out.columns = columns
+        return out
```
Comment (shwina) [inline python/cuspatial/cuspatial/geometry/geodataframe.py:111]: This seems identical to `DataFrame._from_data`, so maybe we just not override it at all?

File: python/cuspatial/cuspatial/geometry/geodataframe.py [hunk 2]
```diff
@@ -90,9 +90,30 @@ def groupby(self, *args, **kwargs):
...
+
 
 class _GeoSeriesUtility:
-    def _from_data(self, new_data, name=None, index=False):
+    @classmethod
+    def _from_data(cls, new_data, name=None, index=False):
```
Comment (shwina) [inline python/cuspatial/cuspatial/geometry/geodataframe.py:116]: 👍

File: python/cuspatial/cuspatial/geometry/geodataframe.py (hunk 1 above)
Comment (vyasr) [reply python/cuspatial/cuspatial/geometry/geodataframe.py:111]: Looking at the constructor of this class, it seems like this method probably needs some logic like 
```
if not isinstance(next(iter(data.values())), GeoColumn):
    # Convert cudf Columns to GeoColumns in `data`
return super()._from_data(data, index, columns)
```
unless this method is never called with a dict of `GeoColumn`s.

**Review by vyasr — COMMENTED** [review]
I was very confused by the original description of this PR since `_from_data` was always a `classmethod` :) I see the edit, but perhaps the change you're referring to happened on a different PR of mine? AFAICT the one you're linking to didn't modify the relevant APIs at all.

**Comment by vyasr** [comment]
I'm aiming to post a fix for this issue tomorrow.

**Pull request closed without merging**